src/containers/b2b/RequisitionPageMain.tsx

```diff
@@ -210,74 +490,89 @@ class RequisitionPageMain extends Component<RequisitionPageMainProps, Requisitio
                   <ArrowLeft className="arrow-left-icon" />
                 </button>
               </div>
             </div>
-            <div className={`product-table ${multiSelectMode ? 'multi-select-mode' : ''}`}>
-              <div className="product-table-heading">
-                <div className="product-table-heading-item" />
-                <div className="product-table-heading-item">
-                  <span>{intl.get('product')}</span>
-                </div>
-                <div className="product-table-heading-item" />
-                <div className="product-table-heading-item">
-                  <span>{intl.get('quick-order-sku-title')}</span>
-                </div>
-                <div className="product-table-heading-item">
-                  <span>{intl.get('quantity')}</span>
-                </div>
-                <div className="product-table-heading-item">
-                  <span>{intl.get('product-options')}</span>
-                </div>
-                <div className="product-table-heading-item">
-                  <span>{intl.get('price')}</span>
-                </div>
-                <div className="product-table-heading-item actions">
-                  <span>{intl.get('actions')}</span>
+            {isPageLoading ? (
+              <div className="loader" />
+            ) : (
+              <div className={`product-table ${multiSelectMode ? 'multi-select-mode' : ''}`}>
+                <div className="product-table-heading">
+                  <div className="product-table-heading-item" />
+                  <div className="product-table-heading-item">
+                    <span>{intl.get('product')}</span>
+                  </div>
+                  <div className="product-table-heading-item" />
+                  <div className="product-table-heading-item">
+                    <span>{intl.get('quick-order-sku-title')}</span>
+                  </div>
+                  <div className="product-table-heading-item">
+                    <span>{intl.get('quantity')}</span>
+                  </div>
+                  <div className="product-table-heading-item">
+                    <span>{intl.get('product-options')}</span>
+                  </div>
+                  <div className="product-table-heading-item">
+                    <span>{intl.get('price')}</span>
+                  </div>
+                  <div className="product-table-heading-item actions">
+                    <span>{intl.get('actions')}</span>
+                  </div>
                 </div>
+                {products.map(product => (
+                  <CartLineItem handleQuantityChange={() => { this.loadRequisitionListData(true); }} item={product} key={product._item[0]._code[0].code} hideAvailabilityLabel isTableView onRemove={() => { this.handleDelete(product); }} onCheck={() => { this.handleCheck(product); }} isChosen={isProductChecked(product)} itemDetailLink="/itemdetail" />
```

Comment:
There's a chance that `product._item[0]` is undefined. Let's try with some of the other products